File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/common/serializedevent/SerializedEvent.java
package cloud.ambar.common.serializedevent;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.Table;
import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Entity
@Table(name = "event_store", indexes = {
        @Index(name = "event_store_idx_event_aggregate_id_version", columnList = "aggregate_id, aggregate_version", unique = true),
        @Index(name = "event_store_idx_event_id", columnList = "event_id", unique = true),
        @Index(name = "event_store_idx_event_causation_id", columnList = "causation_id"),
        @Index(name = "event_store_idx_event_correlation_id", columnList = "correlation_id"),
        @Index(name = "event_store_idx_recorded_on", columnList = "recorded_on"),
        @Index(name = "event_store_idx_event_name", columnList = "event_name")
})
public class SerializedEvent {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty("id")
    private Integer id;

    @Column(name="event_id")
    @JsonProperty("event_id")
    private String eventId;

    @Column(name="aggregate_id")
    @JsonProperty("aggregate_id")
    private String aggregateId;

    @Column(name="causation_id")
    @JsonProperty("causation_id")
    private String causationId;

    @Column(name="correlation_id")
    @JsonProperty("correlation_id")
    private String correlationId;

    @Column(name="aggregate_version")
    @JsonProperty("aggregate_version")
    private Integer aggregateVersion;

    @Column(name="json_payload")
    @JsonProperty("json_payload")
    private String jsonPayload;

    @Column(name="json_metadata")
    @JsonProperty("json_metadata")
    private String jsonMetadata;

    @Column(name="recorded_on")
    @JsonProperty("recorded_on")
    private String recordedOn;

    @Column(name="event_name")
    @JsonProperty("event_name")
    private String eventName;
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCardEnrollment/Common/Events/SerializedEvent.cs
using System.Text.Json.Serialization;

namespace CreditCardEnrollment.Common.Events;

public class SerializedEvent
{
    [JsonPropertyName("id")]
    public int? Id { get; init; }

    [JsonPropertyName("event_id")]
    public required string EventId { get; init; }

    [JsonPropertyName("aggregate_id")]
    public required string AggregateId { get; init; }

    [JsonPropertyName("causation_id")]
    public required string CausationId { get; init; }

    [JsonPropertyName("correlation_id")]
    public required string CorrelationId { get; init; }

    [JsonPropertyName("aggregate_version")]
    public int AggregateVersion { get; init; }

    [JsonPropertyName("json_payload")]
    public required string JsonPayload { get; init; }

    [JsonPropertyName("json_metadata")]
    public string? JsonMetadata { get; init; }

    [JsonPropertyName("recorded_on")]
    public required string RecordedOn { get; init; }

    [JsonPropertyName("event_name")]
    public required string EventName { get; init; }
}
